### PR TITLE
Removes `jam`

### PR DESCRIPTION
**Pull request by ForestEckhardt**
- Removes `jam` from `cargo`
- Removes `scripts` directory as the only scripts was a packaging script
for `jam`
- Removes `jam` packaging from release workflow

Addresses part of #207 

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by fg-j** [comment]
Hey @ForestEckhardt - is there a specific reason why this is still in draft?

**Comment by ForestEckhardt** [comment]
@fg-j I am trying to follow the steps I laid out in issue #207. I would like to have the new `jam` repo completely up and running before I remove the code from this repo. My hope is to have this done by the end of the week but we shall see if that dream boat holds water.

File: README.md [hunk 1]
```diff
@@ -1,15 +1,14 @@
 # packit
 
-Checkout the Go Docs for details on how to use the sub-packages within packit:
-[![GoDoc](https://img.shields.io/badge/pkg.go.dev-doc-blue)](http://pkg.go.dev/github.com/paketo-buildpacks/packit)
+[![GoDoc](https://img.shields.io/badge/pkg.go.dev-doc-blue)](http://pkg.go.dev/.)
```
Comment (ryanmoran) [inline README.md:3]:
```suggestion
[![GoDoc](https://img.shields.io/badge/pkg.go.dev-doc-blue)](http://pkg.go.dev/github.com/paketo-buildpacks/packit)
```

**Comment by ryanmoran** [comment]
The commit history looks pretty wonky. Can you also rebase this?

**Comment by fg-j** [comment]
@ForestEckhardt looks like something funky happened with the commit authors on the most recent commit

File: README.md [hunk 2]
```diff
@@ -225,29 +223,7 @@ the types and functions declared herein.
...
-
-```sh
-jam pack \
-  --buildpack ./buildpack.toml \
-  --stack io.paketo.stacks.tiny \
-  --version 1.2.3 \
-  --offline \
-  --output ./buildpack.tgz
-```
+The `jam` CLI has been moved into its [own dedicated repository](https://github.com/paketo-buildpacks/jam). For new `jam` releases, please visit the `jam` repositories [release page](https://github.com/paketo-buildpacks/jam/releases/tag/v1.0.0).
```
Comment (fg-j) [inline README.md:226]:
```suggestion
The `jam` CLI has been moved into its [own dedicated repository](https://github.com/paketo-buildpacks/jam). For new `jam` releases, please visit the `jam` repositories [release page](https://github.com/paketo-buildpacks/jam/releases).
```

**Comment by fg-j** [comment]
@ForestEckhardt  That would be cool, but I won't block on it.

**Review by fg-j — APPROVED** [review]

**Comment by fg-j** [comment]
@ryanmoran do you have other outstanding concerns here?

**Comment by ryanmoran** [comment]
@fg-j nope. LGTM

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**